newIDE/app/src/MainFrame/ElectronMainMenu.js

```diff
@@ -82,303 +24,69 @@ const useIPCEventListener = (ipcEvent: MainMenuEvent, func) => {
-  useIPCEventListener('main-menu-close-app', props.onCloseApp);
-  useIPCEventListener('main-menu-export', props.onExportProject);
-  useIPCEventListener('main-menu-create-template', props.onCreateProject);
-  useIPCEventListener('main-menu-create-blank', props.onCreateBlank);
+  // We could use a for loop, but for safety let's write every hook one by
+  // one to avoid any change at runtime which would break the rules of hooks.
+  useIPCEventListener('main-menu-open', callbacks.onChooseProject);
+  useIPCEventListener('main-menu-open-recent', callbacks.onOpenRecentFile);
+  useIPCEventListener('main-menu-save', callbacks.onSaveProject);
+  useIPCEventListener('main-menu-save-as', callbacks.onSaveProjectAs);
+  useIPCEventListener('main-menu-close', callbacks.onCloseProject);
+  useIPCEventListener('main-menu-close-app', callbacks.onCloseApp);
+  useIPCEventListener('main-menu-export', callbacks.onExportProject);
+  useIPCEventListener('main-menu-create-template', callbacks.onCreateProject);
+  useIPCEventListener('main-menu-create-blank', callbacks.onCreateBlank);
   useIPCEventListener(
     'main-menu-open-project-manager',
-    props.onOpenProjectManager
+    callbacks.onOpenProjectManager
   );
-  useIPCEventListener('main-menu-open-home-page', props.onOpenHomePage);
-  useIPCEventListener('main-menu-open-debugger', props.onOpenDebugger);
-  useIPCEventListener('main-menu-open-about', props.onOpenAbout);
-  useIPCEventListener('main-menu-open-preferences', props.onOpenPreferences);
-  useIPCEventListener('main-menu-open-language', props.onOpenLanguage);
-  useIPCEventListener('main-menu-open-profile', props.onOpenProfile);
+  useIPCEventListener('main-menu-open-home-page', callbacks.onOpenHomePage);
+  useIPCEventListener('main-menu-open-debugger', callbacks.onOpenDebugger);
+  useIPCEventListener('main-menu-open-about', callbacks.onOpenAbout);
+  useIPCEventListener(
+    'main-menu-open-preferences',
+    callbacks.onOpenPreferences
+  );
+  useIPCEventListener('main-menu-open-language', callbacks.onOpenLanguage);
+  useIPCEventListener('main-menu-open-profile', callbacks.onOpenProfile);
   useIPCEventListener(
     'main-menu-open-games-dashboard',
-    props.onOpenGamesDashboard
+    callbacks.onOpenGamesDashboard
   );
-  useIPCEventListener('update-status', props.setElectronUpdateStatus);
+  useIPCEventListener('update-status', callbacks.setElectronUpdateStatus);
 
   React.useEffect(
     () => {
-      buildAndSendMenuTemplate(project, i18n, recentProjectFiles, shortcutMap);
+      if (ipcRenderer) {
+        ipcRenderer.send(
+          'set-main-menu',
+          buildMainMenuDeclarativeTemplate({
+            project,
+            i18n,
+            recentProjectFiles,
+            shortcutMap,
+            isApplicationTopLevelMenu: true,
+          })
+        );
+      }
     },
     [i18n, language, project, recentProjectFiles, shortcutMap]
   );
 
+  // TODO: move this? And ensure stability.
```

Comment:
not sure about what this comment means.
If it's important to be done later, I'd make the message clearer.

Reply:
Removed the comment. We might at some point move the command palette commands to open recent files, but as this make less sense for now in the web-app, not a big deal.